core/registry.js

```diff
@@ -62,17 +63,21 @@ Blockly.registry.Type.FIELD = new Blockly.registry.Type('field');
 /** @type {!Blockly.registry.Type<Blockly.IToolbox>} */
 Blockly.registry.Type.TOOLBOX = new Blockly.registry.Type('toolbox');
 
+/** @type {!Blockly.registry.Type<Blockly.utils.toolbox.Category>} */
+Blockly.registry.Type.TOOLBOX_CATEGORY = new Blockly.registry.Type('category');
```

Comment:
Since this is a generic registry. I feel category is not specific enough. How about: 
`toolboxCategory` or `toolbox.category`?